Escape % in message and values

msabramo opened the pull request:
Before:

```
... URL='http://192.168.223.128:8083/api/v1/networkpolicies/mattapp%!F(MISSING)mattimage'
```

After:

```
... URL='http://192.168.223.128:8083/api/v1/networkpolicies/mattapp%2Fmattimage'
```

Fixes: https://github.com/adobe-community/issues-ethos/issues/4284

Cc: @iderdik, @jimmyislive

msabramo committed:
Escape % in message and values

Fixes: https://github.com/adobe-community/issues-ethos/issues/4284

iderdik commented on formatters/kv.go:
```diff
@@ -19,9 +19,18 @@ func KVEntryString(entry *logrus.Entry) string {
 	}
 
+	// Escape % if it appears in value, so Sprintf doesn't try to expand it
+	entry.Message = strings.Replace(entry.Message, "%", "%%", -1)
```
what if  there are already 2? This seems like an inherent flaw in sprintf

msabramo replied:
I guess we could add a line before this that converts `"%%"` to `"{double_percent}"` 😄  ?

iderdik replied:
Or we could just use string concatenation. Is there a downside to that?

msabramo replied:
Oh that's a good idea.

iderdik commented:
👍

msabramo committed:
Use concatentation instead of Sprintf

to avoid having to escape percent characters

but we still should escape single quote characters. Otherwise, values
get cut off in Splunk.

And now with tests!

msabramo commented:
@iderdik: 5df5c4e changes to use string concatenation. And I added tests! 😄 

I know you already thumbed it, but since I changed it quite a bit, want to give it another quick once over and make sure I didn't miss anything?

msabramo commented:
Hmmm I don't seem to have the ability to merge.

msabramo commented:
Thanks, @iderdik!
